### PR TITLE
Prevent other users from removing registered snap

### LINKED ISSUE #646: Organisations: Prevent other organisation members from disabling snap auto building [2]
When the user wasn't the one that setup an organisation snap, they can't disable it.

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #646

Final UI part of #646. If repo has registered name user needs to sign in before removing it.
Only user who registered the name is able to remove given repo.

screen shot 2017-05-11 at 14 26 36
screen shot 2017-05-11 at 14 27 32

### CODE REVIEW

File: src/common/components/repository-row/dropdowns/remove-repo-dropdown.js [hunk 1]
```diff
@@ -6,40 +6,102 @@ import { WarningIcon } from '../icons';
 
 import styles from './dropdowns.css';
 
+const getRemoveWarningMessage = (isBuilt, registeredName) => {
```
Comment (bartaz) [inline src/common/components/repository-row/dropdowns/remove-repo-dropdown.js]: Moved from `RepositoryRow` as it makes much more sense now to keep all remove-related logic in one place.

File: src/common/components/repository-row/index.js [hunk 2]
```diff
@@ -281,7 +281,7 @@ export class RepositoryRowView extends Component {
 
     const registeredName = snap.storeName;
 
-    const hasBuilt = !!(latestBuild && snap.snapcraftData);
+    const isBuilt = !!(latestBuild && snap.snapcraftData);
```
Comment (earnubs) [inline src/common/components/repository-row/index.js]: I think this is my fault, but would this be clearer as:

```
(latestBuild && typeof snap.snapcraftData !== 'undefined')
```

Comment (bartaz) [inline src/common/components/repository-row/index.js]: I think `snapcraftData` is (or may be) `null` if snapcraft.yaml is not present.
So I guess `(latestBuild && typeof snap.snapcraftData !== null)`... unless we need to support both `null` and `undefined`...

Comment (earnubs) [inline src/common/components/repository-row/index.js]: Does it need to be `null`, could it be reset to `undefined`?

Comment (bartaz) [inline src/common/components/repository-row/index.js]: I guess it comes all way down from backend handler that reads snapcraft.yaml from GH: https://github.com/canonical-ols/build.snapcraft.io/blob/master/src/server/handlers/github.js#L173

So it seems it's never `undefined`.

File: test/unit/src/common/components/repository-row/dropdowns/t_remove-repo-dropdown.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  let view;
+
+  beforeEach(function() {
+    view = shallow(<RemoveRepoDropdown { ...props }/>);
+  });
+
+  context('when name is not registered', function() {
+    beforeEach(() => {
+      view.setProps({ registeredName: null });
```
Comment (katiefenn) [inline test/unit/src/common/components/repository-row/dropdowns/t_remove-repo-dropdown.js]: I wish I knew about `setProps` earlier... 😅

**Comment by katiefenn** [comment]
+1

File: test/unit/src/common/components/repository-row/dropdowns/t_remove-repo-dropdown.js (hunk 3 above)
Comment (bartaz) [inline test/unit/src/common/components/repository-row/dropdowns/t_remove-repo-dropdown.js]: Me too ;)

**Pull request merged**